pkg/reconciler/taskrun/validate_resources.go

```diff
@@ -88,15 +90,27 @@ func validateParams(paramSpecs []v1beta1.ParamSpec, params []v1beta1.Param) erro
 	if len(missingParamsNoDefaults) > 0 {
 		return fmt.Errorf("missing values for these params which have no default values: %s", missingParamsNoDefaults)
 	}
-	extraParams := list.DiffLeft(providedParams, neededParams)
-	if len(extraParams) != 0 {
-		return fmt.Errorf("didn't need these params but they were provided anyway: %s", extraParams)
+	// If alpha features are enabled, disable checking for extra params.
+	// Extra params are needed to support
+	// https://github.com/tektoncd/community/blob/main/teps/0023-implicit-mapping.md
+	// So that parent params can be passed down to subresources (even if they
+	// are not explicitly used).
+	if config.FromContextOrDefaults(ctx).FeatureFlags.EnableAPIFields != "alpha" {
+		extraParams := list.DiffLeft(providedParams, neededParams)
+		if len(extraParams) != 0 {
+			return fmt.Errorf("didn't need these params but they were provided anyway: %s", extraParams)
+		}
```

Comment:
Restored (see comment below). Added a comment that mirrors the PipelineRun wording, since that seems to have the clearest user benefit rather than just and implementation detail of this feature.